src/services/uploader/uploading.controller.ts

```diff
@@ -103,28 +105,31 @@ export class UploadingController {
   @Post('start_encode')
   async startEncode(@Body() body: StartEncodeDto, @Request() req) {
     const request = parseAndValidateRequest(req, this.#logger);
-    const hiveUsername =
-      body.username || (request.user.network === 'hive' ? request.user.username : undefined);
+    const { account } = parseSub(request.user.sub);
+    const hiveUsername = body.username || (request.user.network === 'hive' ? account : undefined);
     if (!hiveUsername) {
       throw new BadRequestException(
         'Must be signed in with a hive account or include a linked hive account in the request',
       );
     }
+
+    const user = await this.authService.getUserByUserId({ user_id: request.user.user_id });
+
+    if (!user) {
+      throw new UnauthorizedException('User not found');
+    }
+
     if (
-      !(await this.hiveService.isHiveAccountLinked(request.user.sub, hiveUsername)) &&
-      !(request.user.username === hiveUsername && request.user.network === 'hive')
+      !(await this.hiveService.isHiveAccountLinked({
+        account: hiveUsername,
+        user_id: user._id,
+      })) &&
+      !(account === hiveUsername && request.user.network === 'hive')
     ) {
       throw new UnauthorizedException('Your account is not linked to the requested hive account');
     }
-    const accountDetails = await this.hiveChainRepository.getAccount(hiveUsername);
-    if (!accountDetails) throw new NotFoundException('Hive account could not be found');
-    // Check 1: Do we have posting authority?
-    if (this.hiveChainRepository.verifyPostingAuth(accountDetails) === false) {
-      const reason = `Hive Account @${hiveUsername} has not granted posting authority to @threespeak`;
-      const errorType = 'MISSING_POSTING_AUTHORITY';
-      throw new HttpException({ reason: reason, errorType: errorType }, HttpStatus.FORBIDDEN);
-    }
```

Comment:
This should not be removed, as this is when posting authority is required. Users should not be required to give posting authority to log in to Acela Core.

Reply:
Hive chain will just throw this same error if we don't have authority, What is the point of wasting an extra request checking?